src/interface/src/app/map/map.component.ts

```diff
@@ -63,14 +63,16 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     }
 
   ngAfterViewInit(): void {
+    var region: Region|null = null;
     this.initMap();
     this.selectedRegion$.subscribe((selectedRegion) => {
       this.displayRegionBoundary(selectedRegion);
+      region = selectedRegion;
     });
-    this.boundaryService.getHUC12BoundaryShapes().pipe(take(1)).subscribe((boundary: GeoJSON.GeoJSON) => {
+    this.boundaryService.getHUC12BoundaryShapes(region).pipe(take(1)).subscribe((boundary: GeoJSON.GeoJSON) => {
```

Comment:
These calls could happen before region is loaded since region is asynchronous. We can either put these calls that use selectedRegion in the subscribe (but you'll get nested subscribes, which is not recommended) or:

```ts
this.selectedRegion$.pipe(take(1)).subscribe((selectedRegion) => {
  this.displayRegionBoundary(selectedRegion);
});

this.selectedRegion$.pipe(
  take(1),
  switchMap((selectedRegion) => {
    return this.boundaryService.getHUC12BoundaryShapes(selectedRegion).pipe(take(1));
}).subscribe((boundary) => {
    this.initHUC12BoundaryLayer(boundary);
});

this.selectedRegion$.pipe(
  take(1),
  switchMap((selectedRegion) => {
    return this.boundaryService.getCountyBoundaryShapes(selectedRegion).pipe(take(1));
}).subscribe((boundary) => {
    this.initCountyBoundaryLayer(boundary);
});
```
^ switchMap switches to a new observable (region -> boundary)
https://www.thinktecture.com/en/angular/rxjs-antipattern-1-nested-subs/

Reply:
Excellent!  I know just enough to break things.  Do you want to fix this, or should I try?

Reply:
I can update this part!